src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -779,13 +787,8 @@ CHIP_ERROR TimeSynchronizationServer::SetUTCTime(EndpointId ep, uint64_t utcTime
         return err;
     }
     GetDelegate()->UTCTimeAvailabilityChanged(utcTime);
-    mGranularity  = granularity;
-    Status status = TimeSource::Set(ep, source);
-    if (!(status == Status::Success || status == Status::UnsupportedAttribute))
-    {
-        ChipLogError(Zcl, "Writing TimeSource failed.");
-        return CHIP_IM_GLOBAL_STATUS(Failure);
-    }
+    mGranularity = static_cast<GranularityEnum>(to_underlying(granularity) - 1);
```

Comment:
I guess the other PR's strict "greater than" check is what keeps this from running off the bottom of the list?